src/lib.rs

```diff
@@ -99,33 +124,67 @@ pub fn run(args: Args) {
             );
 
             get_searcher(output_mode)
+                .borrow_mut()
                 .search_path(&matcher, path, printer.sink_with_path(&matcher, path))
                 .unwrap();
+            buffer_writer.print(printer.get_mut()).unwrap();
         });
 }
 
-fn format_relative_path(path: &Path, output_mode: OutputMode) -> &Path {
-    if output_mode == OutputMode::Vimgrep && path.starts_with("./") {
-        path.strip_prefix("./").unwrap()
-    } else {
-        path
-    }
+type Printer = Standard<Buffer>;
+
+thread_local! {
+    static PRINTER: OnceCell<(Rc<RefCell<Printer>>, OutputMode)> = Default::default();
+}
+fn get_printer(buffer_writer: &BufferWriter, output_mode: OutputMode) -> Rc<RefCell<Printer>> {
+    PRINTER.with(|printer| {
+        let (printer, output_mode_when_initialized) = printer.get_or_init(|| {
+            (
+                Rc::new(RefCell::new(create_printer(buffer_writer, output_mode))),
+                output_mode,
+            )
+        });
+        assert!(
+            *output_mode_when_initialized == output_mode,
+            "Using multiple output modes not supported"
+        );
+        printer.clone()
+    })
 }
 
-fn get_printer(output_mode: OutputMode) -> grep::printer::Standard<termcolor::NoColor<io::Stdout>> {
+fn create_printer(buffer_writer: &BufferWriter, output_mode: OutputMode) -> Printer {
     match output_mode {
-        OutputMode::Normal => grep::printer::Standard::new_no_color(io::stdout()),
```

Comment:
I got rid of the `*_no_color()` because it seemed like that's actually the default and it was making `printer.get_mut()` (above) not directly return a `Buffer` (because specifically I think `*_no_color()` would return a `Standard<NoColor<Buffer>>` instead of just a `Standard<Buffer>`)